tests/test_regression.py

```diff
@@ -1,100 +1,335 @@
-            h.update(f.read())
-        if h.hexdigest() != r["sha1-checksum"]:
-            raise RuntimeError("They tooks our data!!! All our rackspace are belong to them.")
-        sim_files[r["input-file"]] = fpath
-
-class TestRegression(object):
-    def __init__(self):
-        self.in_dir_ = "../input"
-        self.tmp_files_ = {}        
-        for root, dirs, files in os.walk(self.in_dir_):
-            for f in files:
-                self.tmp_files_[f] = str(uuid.uuid4()) + ".h5"
+import numpy as np
+from numpy.testing import assert_array_almost_equal 
+from nose.tools import assert_equal, assert_true
+
+from helper import check_cmd, run_cyclus, table_exist, find_ids
+
+class TestRegression(TestCase):
+    """A base class for all regression tests. A derived class is required for
+    each new input file to be tested. Each derived class *must* declare an `inf`
+    member in their `__init__` function that points to the input file to be
+    tested, e.g., `self.inf_ = ./path/to/my/input_file.xml. See below for
+    examples.
+    """
+    def __init__(self, *args, **kwargs):
+        super(TestRegression, self).__init__(*args, **kwargs)
+        self.outf = str(uuid.uuid4()) + ".h5"
+        self.inf = None
 
     def __del__(self):
-        for inf, outf in self.tmp_files_.items():
-            if os.path.isfile(outf):
-                print("removing {0}".format(outf))
-                os.remove(outf)
-                
-    def teardown(self):
-        for inf, outf in self.tmp_files_.items():
-            if os.path.isfile(outf):
-                print("removing {0}".format(outf))
-                os.remove(outf)
-
-    def test_regression(self, check_deterministic=False):
-        """Test for all inputs in sim_files. Checks if reference and current cyclus 
-        output is the same.
-
-        Parameters
-        ----------
-        check_deterministic : bool
-            If True, also test determinisitc equality of simulations
+        if os.path.isfile(self.outf):
+            print("removing {0}".format(self.outf))
+            os.remove(self.outf)
+
+    def setUp(self):
+        if not self.inf:
+            raise TypeError(("self.inf must be set in derived classes "
+                             "to run regression tests."))
+        run_cyclus("cyclus", os.getcwd(), self.inf, self.outf)        
+
+        with tables.open_file(self.outf, mode="r") as f:
+            # Get specific tables and columns
+            self.agent_entry = f.get_node("/AgentEntry")[:]
+            self.agent_exit = f.get_node("/AgentExit")[:] if "/AgentExit" in f \
+                else None
+            self.resources = f.get_node("/Resources")[:]
+            self.transactions = f.get_node("/Transactions")[:]
+            self.compositions = f.get_node("/Compositions")[:]
+            self.info = f.get_node("/Info")[:]
+            self.rsrc_qtys = {
+                x["ResourceId"]: x["Quantity"] for x in self.resources}
```

Comment:
x here is equivalent to a row. Thus for every row, I am mapping the
resource id to the quantity.

On Wed, Jan 7, 2015 at 11:34 AM, Anthony Scopatz <notifications@github.com>
wrote:

> In tests/test_regression.py
> <https://github.com/cyclus/cycamore/pull/296#discussion-diff-22600846>:
>
> > +        if not self.inf:
> > +            raise TypeError(("self.inf must be set in derived classes "
> > +                             "to run regression tests."))
> > +        run_cyclus("cyclus", os.getcwd(), self.inf, self.outf)
> > +
> > +        with tables.open_file(self.outf, mode="r") as f:
> > +            # Get specific tables and columns
> > +            self.agent_entry = f.get_node("/AgentEntry")[:]
> > +            self.agent_exit = f.get_node("/AgentExit")[:] if "/AgentExit" in f \
> > +                else None
> > +            self.resources = f.get_node("/Resources")[:]
> > +            self.transactions = f.get_node("/Transactions")[:]
> > +            self.compositions = f.get_node("/Compositions")[:]
> > +            self.info = f.get_node("/Info")[:]
> > +            self.rsrc_qtys = {
> > +                x["ResourceId"]: x["Quantity"] for x in self.resources}
>
> Here, because it is a value in a dict, x["Quantity"] for x in
> self.resources will evaluate to self.resources[-1]["Quantity"]. That is,
> the last quantity. Is this the intended behaviour?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/296/files#r22600846>.
>



-- 
Matthew Gidden
Ph.D. Candidate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192